samcli/lib/providers/provider.py

```diff
@@ -858,5 +893,6 @@ def _get_build_dir(resource: Union[Function, LayerVersion], build_root: str) ->
     """
     Return the build directory to place build artifact
     """
+    resource_id = resource.function_id if isinstance(resource, Function) else resource.layer_id
```

Comment:
I would prefer to find ways to leave the default to be `resource.name`. This is a much more maintainable solution. For example, if we add a new builder this will try to access `resource.layer_id` instead of "just working". If we need to change this, I am ok but let's change this to check for each resource and access the attribute we need. Otherwise default to `resource.name`.

Reply:
You are correct.
This time, we noticed that the value of resource.name coming from the CDK output template has been changed, and the name of the generated build directory has changed. Therefore, we have included this workaround for backward compatibility. If you are not affected by changing the name of the build directory, you can use the resource.name as it is. If there is no guarantee of this, then resource.name is the default.